app/services/proxy_status.rb

```diff
@@ -26,6 +26,18 @@ def tftp_server
     end
   end
 
+  def environment_stats
```

Comment:
tested on about 200 classes and 3 environments, this runs in about 2 seconds, so for now I closed the smart proxy pr, will reopen it if we see this is getting to be a performance bottleneck.